compiler-cli/src/new.rs

```diff
@@ -48,8 +58,22 @@ impl Creator {
         .trim()
         .to_string();
 
+        let files_to_create = HashMap::from([
```

Comment:
If we want to use types to model the files to create let's use this pattern here:

https://github.com/gleam-lang/gleam/blob/main/compiler-core/src/type_/prelude.rs#L180

We could then implement a `pub fn location(&self) -> PathBuf` which would be the replacement for the hashmap.

The use of EnumIter means that we are guaranteed to cover each of the variants in the enum and we don't have to have the runtime checks to ensure the entries are in the hashmap.

Reply:
I gave it a shot. Let me know if this is the right track :)